twisted/pydoctor
# Build pydoctor apidocs together with Sphinx.

Scope
=====

This add a Sphinx extension that will trigger pydoctor apidocs build together with sphinx build.

Changes
=======

Add a pydoctor extension that is called at the end of the build.

pydoctor warnings are converted into sphinx warnings.


Follow up work
===========

1. Once this is merge, we can have a public pydoctor sphinx inventory object hosted by Read The Docs and use intersphinx to link from Sphinx to pydoctor  API (See also 6)

2. We need to check how this behaves on RTD tags / stable / latest builds.

3. We should also check RTD stabe vs latest and decide which is which.
* https://pydoctor.readthedocs.io/en/latest/
* https://pydoctor.readthedocs.io/en/stable/

4. Fix all pydoctor warnings from inside the pydoctor own code #269

5. Create some "pydoctor RTD" theme that will provide better integrating with the RTD theme.

6 Build pydoctor intersphinx object before calling RTD sphinx build so that we can reference new pydoctor API included in the branch.


How to test
=========

```
tox -e docs
firefox build/docs/index.html
```

On RTD you can use https://pydoctor--254.org.readthedocs.build/en/254/

Intersphinx generated at https://pydoctor--254.org.readthedocs.build/en/254/api/objects.inv

Check the magic inside the "Quicklinks" TOC.
